llama-index-core/tests/workflow/test_decorator.py

```diff
@@ -0,0 +1,23 @@
+import pytest
+
+from llama_index.core.workflow.decorators import step
+from llama_index.core.workflow.events import Event
+
+
+def test_decorated_config():
+    def f(self, ev: Event) -> Event:
+        return Event()
+
+    res = step()(f)
+    config = getattr(res, "__step_config")
+    assert config.accepted_events == [Event]
+    assert config.event_name == "ev"
+    assert config.return_types == [Event]
+
+
+def test_decorate_wrong_signature():
+    def f():
+        pass
+
+    with pytest.raises(ValueError):
```

Comment:
This actually raises a WorkflowValidationError now (I meant to push the change, but need permissions 😉)